### ISSUE TITLE
Modem device not there after wakeup

### ISSUE THREAD

**Issue by maricn** [juhovh/xmm7360_usb#10]
I set it up per README.md, and I can see the modem and use it for internet and also SMS messages. However that works only since the boot until the first sleep. After that, the device is not there anymore.

I'm sorry if this is not a problem with your module, but I thought it's the best place to ask.

I'm on Thinkpad x1c7 (X1 Carbon 7th Gen).
```
$ uname -a
Linux work-x1c-mx19 5.6.0-18.1-liquorix-amd64 #1 ZEN SMP PREEMPT liquorix 5.6-20.1~buster (2020-06-10) x86_64 GNU/Linux

$ sudo lsmod | grep xmm
xmm7360_usb            16384  0

$ sudo lsusb
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 06cb:00bd Synaptics, Inc.
Bus 001 Device 002: ID 04f2:b67c Chicony Electronics Co., Ltd
Bus 001 Device 004: ID 8087:0aaa Intel Corp.
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

$ sudo lspci -nnvd 8086:7360


$ sudo mmcli -m 0
error: couldn't find modem at '/org/freedesktop/ModemManager1/Modem/0'

$ sudo mmcli -L
No modems were found

$ sudo mmcli -S
error: couldn't request to scan devices: 'GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._g_2dio_2derror_2dquark.Code19: Operation was cancelled'
```

[dmesg.log](https://github.com/juhovh/xmm7360_usb/files/4795037/dmesg.log)

Also, `lshw` doesn't seem to have anything relevant, and `/dev/ttyACM*` are gone.

Thanks for any help!

**Comment by maricn** [comment]
Just rebooted it, this is how it looks like:
```
  --------------------------------
  General  |            dbus path: /org/freedesktop/ModemManager1/Modem/0
           |            device id: 1cd00d6cd824a4f8e09bd9c7bd***********
  --------------------------------
  Hardware |         manufacturer: Fibocom
           |                model: MBIM + 3 CDC-ACM
           |             revision: 18500.5001.00.01.20.75
           |         h/w revision: V1.0.4
           |            supported: gsm-umts, lte
           |              current: gsm-umts, lte
           |         equipment id: 01555000*******
  --------------------------------
  System   |               device: /sys/devices/pci0000:00/0000:00:14.0/usb1/1-7
           |              drivers: cdc_acm, cdc_mbim
           |               plugin: Fibocom
           |         primary port: cdc-wdm0
           |                ports: cdc-wdm0 (mbim), wwan0 (net), ttyACM1 (unknown), 
           |                       ttyACM2 (at), ttyACM0 (at)
  --------------------------------
  Numbers  |                  own: +49176*******
  --------------------------------
  Status   |       unlock retries: sim-pin2 (3)
           |                state: connected
           |          power state: on
           |          access tech: lte
           |       signal quality: 12% (cached)
  --------------------------------
  Modes    |            supported: allowed: 3g; preferred: none
           |                       allowed: 4g; preferred: none
           |                       allowed: 3g, 4g; preferred: none
           |                       allowed: 3g, 4g; preferred: 3g
           |                       allowed: 3g, 4g; preferred: 4g
           |              current: allowed: 3g, 4g; preferred: 4g
  --------------------------------
  Bands    |            supported: utran-1, utran-4, utran-5, utran-8, utran-2, eutran-1, 
           |                       eutran-2, eutran-3, eutran-4, eutran-5, eutran-7, eutran-8, 
           |                       eutran-12, eutran-13, eutran-17, eutran-18, eutran-19, eutran-20, 
           |                       eutran-26, eutran-28, eutran-29, eutran-30, eutran-41, eutran-66
           |              current: utran-1, utran-4, utran-5, utran-8, utran-2, eutran-1, 
           |                       eutran-2, eutran-3, eutran-4, eutran-5, eutran-7, eutran-8, 
           |                       eutran-12, eutran-13, eutran-17, eutran-18, eutran-19, eutran-20, 
           |                       eutran-26, eutran-28, eutran-29, eutran-30, eutran-41, eutran-66
  --------------------------------
  IP       |            supported: ipv4, ipv6, ipv4v6
  --------------------------------
  3GPP     |                 imei: 01555**********
           |        enabled locks: fixed-dialing
           |          operator id: 26203
           |        operator name: 1&1
           |         registration: home
  --------------------------------
  3GPP EPS | ue mode of operation: csps-2
  --------------------------------
  SIM      |            dbus path: /org/freedesktop/ModemManager1/SIM/0
  --------------------------------
  Bearer   |            dbus path: /org/freedesktop/ModemManager1/Bearer/1
```

**Comment by alembiq** [comment]
same problem here :( everytime i want to use it i need to reboot :( 

Linux 5.6.0-2parrot1-amd64 #1 SMP Debian 5.6.14-2parrot1 (2020-05-23) x86_64 GNU/Linux

**Comment by moulip** [comment]
I have to add my voice to this. The behavior is quite strange. My laptop always remains in suspend mode. And one time in two, network-manager is showing the mobile broadband connection or not. I mean, I open my laptop, no mobile broadband is showing up. I close the lid, it goes in suspend, I wake it up and the mobile connection appears. I can't figure out why.

**Comment by turtleizzy** [comment]
I was not able to use the modem for SMS messages - can you tell me how did you configure to send SMS?

**Comment by maricn** [comment]
> I was not able to use the modem for SMS messages - can you tell me how did you configure to send SMS?

Try installing modem manager GUI. Also, your SIM card should be enabled to send SMS messages. I am not very experienced and can't offer much more help, but I think this is not the right place to ask such questions unless you have something to add on the topic of modem device disappearing after sleep. I can recommend you create another issue, at minimum.

**Comment by turtleizzy** [comment]
Sorry for the irrelevant post. 
My modem device seems to be able to restart properly after waking up from S3 sleep (x1c 6th & linux mint 20 & Fibocom L850-GL modem). Here is a sample dmesg log of sleeping & restarting. I don't know if this may help. 
The driver was from this repo except that I wrote a simple dkms configuration to help compling with kernel in ubuntu 20.04 setting. 
```
[90548.550475] e1000e: enp0s31f6 NIC Link is Down
[90548.667260] wlp2s0: deauthenticating from xx:xx:xx:xx:xx:xx by local choice (Reason: 3=DEAUTH_LEAVING)
[90549.085266] PM: suspend entry (deep)
[90549.092126] Filesystems sync: 0.006 seconds
[90549.092845] Freezing user space processes ... (elapsed 0.003 seconds) done.
[90549.095998] OOM killer disabled.
[90549.095999] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[90549.098106] printk: Suspending console(s) (use no_console_suspend to debug)
[90549.287872] e1000e: EEE TX LPI TIMER: 00000011
[90549.690077] ACPI: EC: interrupt blocked
[90549.749505] ACPI: Preparing to enter system sleep state S3
[90549.757959] ACPI: EC: event blocked
[90549.757961] ACPI: EC: EC stopped
[90549.757962] PM: Saving platform NVS memory
[90549.757974] Disabling non-boot CPUs ...
[90549.760003] smpboot: CPU 1 is now offline
[90549.764490] IRQ 139: no longer affine to CPU2
[90549.765503] smpboot: CPU 2 is now offline
[90549.771099] smpboot: CPU 3 is now offline
[90549.776852] smpboot: CPU 4 is now offline
[90549.781472] IRQ 126: no longer affine to CPU5
[90549.782489] smpboot: CPU 5 is now offline
[90549.787101] IRQ 129: no longer affine to CPU6
[90549.788139] smpboot: CPU 6 is now offline
[90549.793865] smpboot: CPU 7 is now offline
[90549.800907] ACPI: Low-level resume complete
[90549.800993] ACPI: EC: EC started
[90549.800994] PM: Restoring platform NVS memory
[90549.804916] Enabling non-boot CPUs ...
[90549.804969] x86: Booting SMP configuration:
[90549.804970] smpboot: Booting Node 0 Processor 1 APIC 0x2
[90549.809726] CPU1 is up
[90549.809757] smpboot: Booting Node 0 Processor 2 APIC 0x4
[90549.811774] CPU2 is up
[90549.811800] smpboot: Booting Node 0 Processor 3 APIC 0x6
[90549.812436] CPU3 is up
[90549.812464] smpboot: Booting Node 0 Processor 4 APIC 0x1
[90549.813215] CPU4 is up
[90549.813245] smpboot: Booting Node 0 Processor 5 APIC 0x3
[90549.813899] CPU5 is up
[90549.813924] smpboot: Booting Node 0 Processor 6 APIC 0x5
[90549.814590] CPU6 is up
[90549.814616] smpboot: Booting Node 0 Processor 7 APIC 0x7
[90549.815295] CPU7 is up
[90549.817876] ACPI: Waking up from system sleep state S3
[90549.840148] ACPI: EC: interrupt unblocked
[90549.912096] ACPI: EC: event unblocked
[90549.963106] usb 2-3: Disable of device-initiated U1 failed.
[90549.966606] usb 2-3: Disable of device-initiated U2 failed.
[90550.128903] nvme nvme0: Shutdown timeout set to 8 seconds
[90550.141573] nvme nvme0: 8/0/0 default/read/poll queues
[90550.159938] usb 2-3: reset SuperSpeed Gen 1 USB device number 2 using xhci_hcd
[90550.319828] usb 1-9: reset full-speed USB device number 4 using xhci_hcd
[90550.664614] psmouse serio1: synaptics: queried max coordinates: x [..5676], y [..4760]
[90550.699313] psmouse serio1: synaptics: queried min coordinates: x [1266..], y [1094..]
[90551.229120] acpi LNXPOWER:01: Turning OFF
[90551.229761] OOM killer enabled.
[90551.229763] Restarting tasks ... 
[90551.230407] usb 1-6: USB disconnect, device number 16
[90551.230746] xmm7360_usb: Found a PCI device 8086:7360
[90551.230759] xmm7360_usb: Link status changed, trying to reset the device
[90551.235945] cdc_ncm 1-6:1.6 enx000011121314: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[90551.240503] done.
[90551.246722] thermal thermal_zone6: failed to read out thermal zone (-61)
[90551.279652] cdc_ncm 1-6:1.8 enx000011121316: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[90551.281208] PM: suspend exit
[90551.319544] cdc_ncm 1-6:1.10 enx000011121318: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[90551.659816] xmm7360_usb: Device reset successfully
[90551.660311] pci 0000:03:00.0: [8086:7360] type 00 class 0x0d4000
[90551.660362] pci 0000:03:00.0: reg 0x10: [mem 0x00000000-0x00000fff 64bit]
[90551.660382] pci 0000:03:00.0: reg 0x18: [mem 0x00000000-0x000003ff 64bit]
[90551.660532] pci 0000:03:00.0: PME# supported from D0 D3hot D3cold
[90551.660800] pci 0000:03:00.0: BAR 0: assigned [mem 0xee200000-0xee200fff 64bit]
[90551.660819] pci 0000:03:00.0: BAR 2: assigned [mem 0xee201000-0xee2013ff 64bit]
[90551.660880] xmm7360_usb: Found a PCI device 8086:7360
[90551.660884] xmm7360_usb: Link status changed, trying to reset the device
[90552.076103] xmm7360_usb: Device reset successfully
[90552.287500] usb 1-6: new high-speed USB device number 17 using xhci_hcd
[90552.436233] usb 1-6: New USB device found, idVendor=8087, idProduct=07f5, bcdDevice= 0.00
[90552.436239] usb 1-6: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[90552.447170] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
[90553.646795] usb 1-6: USB disconnect, device number 17
[90559.385705] iwlwifi 0000:02:00.0: Applying debug destination EXTERNAL_DRAM
[90559.520794] iwlwifi 0000:02:00.0: Applying debug destination EXTERNAL_DRAM
[90559.590504] iwlwifi 0000:02:00.0: FW already configured (0) - re-configuring
[90559.659014] usb 1-6: new high-speed USB device number 18 using xhci_hcd
[90559.815876] usb 1-6: New USB device found, idVendor=8087, idProduct=095a, bcdDevice= 3.33
[90559.815882] usb 1-6: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[90559.815886] usb 1-6: Product: MODEM + 2 CDC-ACM + 3 CDC-NCM + SS
[90559.815889] usb 1-6: Manufacturer: Intel Corp.
[90559.815891] usb 1-6: SerialNumber: 12345123451234512345
[90559.844370] cdc_acm 1-6:1.0: ttyACM0: USB ACM device
[90559.846613] cdc_acm 1-6:1.2: ttyACM1: USB ACM device
[90559.848629] cdc_acm 1-6:1.4: ttyACM2: USB ACM device
[90559.879478] cdc_ncm 1-6:1.6: MAC-Address: 00:00:11:12:13:14
[90559.879484] cdc_ncm 1-6:1.6: setting rx_max = 16384
[90559.880825] cdc_ncm 1-6:1.6 usb0: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, 00:00:11:12:13:14
[90559.915064] cdc_ncm 1-6:1.8: MAC-Address: 00:00:11:12:13:16
[90559.915072] cdc_ncm 1-6:1.8: setting rx_max = 16384
[90559.916548] cdc_ncm 1-6:1.8 usb1: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, 00:00:11:12:13:16
[90559.951021] cdc_ncm 1-6:1.10: MAC-Address: 00:00:11:12:13:18
[90559.951030] cdc_ncm 1-6:1.10: setting rx_max = 16384
[90559.952504] cdc_ncm 1-6:1.10 usb2: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, 00:00:11:12:13:18
[90560.765263] cdc_ncm 1-6:1.6 enx000011121314: renamed from usb0
[90560.805087] cdc_ncm 1-6:1.8 enx000011121316: renamed from usb1
[90560.837383] cdc_ncm 1-6:1.10 enx000011121318: renamed from usb2
```

**Comment by moulip** [comment]
Hello,

Does it restart properly each time you wake up your laptop ?
To be able to put the laptop in suspend mode without emptying the
battery, I had to tweak the grub default configuration by adding,
mem_sleep_default=deep to the GRUB_CMDLINE_DEFAULT line.
On my laptop the modem only appears one out of two laptop wakings up.

**Comment by turtleizzy** [comment]
I did experience some fluctuations however I was able to restart it by rescanning pci bus with `echo "1" | sudo tee -a /sys/bus/pci/rescan`. Here is a sample dmesg log.

Not recognized as usb device after a wake up.
```
[39915.529291] xmm7360_usb 0000:03:00.0: Refused to change power state, currently in D3
[39915.580107] ACPI: EC: event unblocked
[39915.589937] usb 2-3: Disable of device-initiated U1 failed.
[39915.593571] usb 2-3: Disable of device-initiated U2 failed.
[39915.788535] usb 2-3: reset SuperSpeed Gen 1 USB device number 2 using xhci_hcd
[39915.797228] nvme nvme0: Shutdown timeout set to 8 seconds
[39915.806790] nvme nvme0: 8/0/0 default/read/poll queues
[39915.936719] usb 1-9: reset full-speed USB device number 4 using xhci_hcd
[39916.307837] psmouse serio1: synaptics: queried max coordinates: x [..5676], y [..4760]
[39916.340081] psmouse serio1: synaptics: queried min coordinates: x [1266..], y [1094..]
[39916.846058] acpi LNXPOWER:01: Turning OFF
[39916.846742] OOM killer enabled.
[39916.846743] Restarting tasks ... 
[39916.847393] usb 1-6: USB disconnect, device number 6
[39916.847707] xmm7360_usb: Found a PCI device 8086:7360
[39916.847720] xmm7360_usb: Link status changed, trying to reset the device
[39916.852025] cdc_ncm 1-6:1.6 enx000011121314: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[39916.858458] done.
[39916.875503] thermal thermal_zone6: failed to read out thermal zone (-61)
[39916.888306] cdc_ncm 1-6:1.8 enx000011121316: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[39916.905992] PM: suspend exit
[39916.916397] cdc_ncm 1-6:1.10 enx000011121318: unregister 'cdc_ncm' usb-0000:00:14.0-6, CDC NCM
[39917.269153] xmm7360_usb: Device reset successfully
[39917.472335] usb 1-6: new high-speed USB device number 7 using xhci_hcd
[39917.624962] usb 1-6: New USB device found, idVendor=8087, idProduct=07f5, bcdDevice= 0.00
[39917.624969] usb 1-6: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[39918.053059] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
[39918.836563] usb 1-6: USB disconnect, device number 7
```

Recovered by rescanning pci bus.
```
[41336.549379] pci 0000:03:00.0: [8086:7360] type 00 class 0x0d4000
[41336.549444] pci 0000:03:00.0: reg 0x10: [mem 0x00000000-0x00000fff 64bit]
[41336.549472] pci 0000:03:00.0: reg 0x18: [mem 0x00000000-0x000003ff 64bit]
[41336.549696] pci 0000:03:00.0: PME# supported from D0 D3hot D3cold
[41336.550253] pci 0000:03:00.0: BAR 0: assigned [mem 0xee200000-0xee200fff 64bit]
[41336.550278] pci 0000:03:00.0: BAR 2: assigned [mem 0xee201000-0xee2013ff 64bit]
[41336.550383] xmm7360_usb: Found a PCI device 8086:7360
[41336.550388] xmm7360_usb: Link status changed, trying to reset the device
[41336.970268] xmm7360_usb: Device reset successfully
[41337.137513] usb 1-6: new high-speed USB device number 8 using xhci_hcd
[41337.286624] usb 1-6: New USB device found, idVendor=8087, idProduct=07f5, bcdDevice= 0.00
[41337.286630] usb 1-6: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[41338.495452] usb 1-6: USB disconnect, device number 8
[41344.349477] usb 1-6: new high-speed USB device number 9 using xhci_hcd
[41344.507013] usb 1-6: New USB device found, idVendor=8087, idProduct=095a, bcdDevice= 3.33
[41344.507018] usb 1-6: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[41344.507021] usb 1-6: Product: MODEM + 2 CDC-ACM + 3 CDC-NCM + SS
[41344.507024] usb 1-6: Manufacturer: Intel Corp.
[41344.507026] usb 1-6: SerialNumber: xxxxxxxxxxxx
[41344.534923] cdc_acm 1-6:1.0: ttyACM0: USB ACM device
[41344.536720] cdc_acm 1-6:1.2: ttyACM1: USB ACM device
[41344.538597] cdc_acm 1-6:1.4: ttyACM2: USB ACM device
[41344.570404] cdc_ncm 1-6:1.6: MAC-Address: aa:bb:cc:dd:ee:ff
[41344.570410] cdc_ncm 1-6:1.6: setting rx_max = 16384
[41344.572024] cdc_ncm 1-6:1.6 usb0: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, aa:bb:cc:dd:ee:ff
[41344.605453] cdc_ncm 1-6:1.8: MAC-Address: aa:bb:cc:dd:ee:ff
[41344.605458] cdc_ncm 1-6:1.8: setting rx_max = 16384
[41344.606900] cdc_ncm 1-6:1.8 usb1: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, aa:bb:cc:dd:ee:ff
[41344.641299] cdc_ncm 1-6:1.10: MAC-Address: aa:bb:cc:dd:ee:ff
[41344.641305] cdc_ncm 1-6:1.10: setting rx_max = 16384
[41344.642742] cdc_ncm 1-6:1.10 usb2: register 'cdc_ncm' at usb-0000:00:14.0-6, CDC NCM, aa:bb:cc:dd:ee:ff
[41345.391096] cdc_ncm 1-6:1.6 enx000011121314: renamed from usb0
[41345.409969] cdc_ncm 1-6:1.8 enx000011121316: renamed from usb1
[41345.435070] cdc_ncm 1-6:1.10 enx000011121318: renamed from usb2
```

> Hello, Does it restart properly each time you wake up your laptop ? To be able to put the laptop in suspend mode without emptying the battery, I had to tweak the grub default configuration by adding, mem_sleep_default=deep to the GRUB_CMDLINE_DEFAULT line. On my laptop the modem only appears one out of two laptop wakings up.

**Comment by moulip** [comment]
Hi,

I have just tried your fix and it seems to work. I have been able to
rescan the PCI bus and the modem showed up in Network-Manager.
I will use that for now.
